homeassistant/components/sensor/sht31.py

```diff
@@ -0,0 +1,111 @@
+"""
+Support for Sensirion SHT31 temperature and humidity sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.sht31/
+"""
+
+import logging
+import math
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    TEMP_FAHRENHEIT, CONF_NAME, CONF_MONITORED_CONDITIONS)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util.temperature import celsius_to_fahrenheit
+
+REQUIREMENTS = ['Adafruit-GPIO==1.0.3',
+                'Adafruit-SHT31==1.0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_I2C_ADDRESS = 'i2c_address'
+
+DEFAULT_NAME = 'SHT31'
+DEFAULT_I2C_ADDRESS = '0x44'
+
+SENSOR_TEMPERATURE = 'temperature'
+SENSOR_HUMIDITY = 'humidity'
+SENSOR_TYPES = {
+    SENSOR_TEMPERATURE: ['Temperature', None],
+    SENSOR_HUMIDITY: ['Humidity', '%']
+}
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=SENSOR_TYPES.keys()):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    from Adafruit_SHT31 import SHT31
+
+    SENSOR_TYPES[SENSOR_TEMPERATURE][1] = hass.config.units.temperature_unit
```

Comment:
I know DHT does the same, but this code smells a lot to me. `SENSOR_TYPES` is supposed to be a `constant`, but you're setting values here. Additionally, this won't allow for dynamic changing of temperature units (if it is someday maybe added) and moving the temperature unit checking could easily be moved.

```python
if not math.isnan(temperature):
    self._state = round(temperature.convert(
        temperature, TEMP_CELSIUS, hass.config.units.temperature_unit), 1)
else:
    _LOGGER.warning("Bad sample from sensor %s", self.name)
```

Reply:
Referring strictly to assigning to the constant, I've updated the code to use a lambda, but this just makes the code more complex without much advantage, it's still not dynamic and i'm not proud of the change ;)
Now about the dynamic part, I'm not sure how to implement that. I thought about passing a "hass" reference to the Sensor instance, but I'm not sure about the implications of that (memory leaks?). I would leave it like that just to keep it simple and avoid some over engineering.

Reply:
About the hass instance: `self.hass` will automatically be set for each entity (*). So that part could be:

```python
round(temperature.convert(temperature, TEMP_CELSIUS, self.hass.config.units.temperature_unit), 1)
```

* If it is part of a so-called "PlatformComponent" + only after `async_added_to_hass` has been called, but that's usually not a concern.